src/dra/7879C.c

```diff
@@ -2364,32 +2364,34 @@ void EntityPlayerBlinkWhite(Entity* self) {
     Entity* sp3C;
     s32 sp38;
 
-    sp70 = (self->params & 0x7F00) >> 8;
+    upperParams = (self->params & 0x7F00) >> 8;
     sp48 = 0;
-    if ((PLAYER.animSet == 0xF) && (sp70 == 0x23) && g_Player.unk66 < 2) {
+    if ((PLAYER.animSet == 0xF) && (upperParams == 0x23) && g_Player.unk66 < 2) {
         sp48 = 1;
     }
-    if ((((sp70 & 0x3F) != 0x1D) &&
+    if ((((upperParams & 0x3F) != 0x1D) &&
          (g_Player.status & PLAYER_STATUS_MIST_FORM)) ||
         (g_Player.status & PLAYER_STATUS_AXEARMOR)) {
-        goto block_229;
+        goto set_unk6C_unk6E_0;
     }
-    if ((g_Player.unk6C) && sp70 != 0x20 && sp70 != 0x21 &&
-        ((sp70 & 0x3F) != 0x1D)) {
+    if ((g_Player.unk6C) && upperParams != 0x20 && upperParams != 0x21 &&
+        ((upperParams & 0x3F) != 0x1D)) {
         g_Player.unk6E = 0;
-        goto block_231;
+        DestroyEntity(self);
+        return;
     }
-    if ((g_Player.unk6E) && sp70 != 0x23 && sp70 != 0x24 &&
-        ((sp70 & 0x3F) != 0x1D)) {
+    if ((g_Player.unk6E) && upperParams != 0x23 && upperParams != 0x24 &&
+        ((upperParams & 0x3F) != 0x1D)) {
         g_Player.unk6C = 0;
-        goto block_231;
+        DestroyEntity(self);
+        return;
```

Comment:
Did not look too closely but I think removing these gotos breaks the PSP match?
You might need to bring in your changes to the PSP scratch and play around with it some more
https://decomp.me/scratch/GoMQZ